code/game/objects/items/weapons/storage/boxes.dm

```diff
@@ -338,6 +339,62 @@
 	new /obj/item/weapon/grenade/flashbang(src)
 	new /obj/item/weapon/grenade/flashbang(src)
 
+/obj/item/weapon/storage/box/firingpins
+	name = "box of firing pins"
+	desc = "A box of NT brand Firearm authentication devices - Needed to operate most weapons."
+
+/obj/item/weapon/storage/box/firingpins/fill()
+	..()
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin(src)
+	name = "box of firing pins"
+	desc = "A box of NT brand Firearm authentication devices - Needed to operate most weapons."
+
+/obj/item/weapon/storage/box/firingpins/fill()
+	..()
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin(src)
+
+/obj/item/weapon/storage/box/testpins
+	name = "box of firing pins"
+	desc = "A box of NT brand Testing Authentication pins, allows guns to fire in designated areas."
+
+/obj/item/weapon/storage/box/firingpins/test/fill()
+	..()
+	new /obj/item/device/firing_pin/test_range(src)
+	new /obj/item/device/firing_pin/test_range(src)
+	new /obj/item/device/firing_pin/test_range(src)
+	new /obj/item/device/firing_pin/test_range(src)
+	new /obj/item/device/firing_pin/test_range(src)
+	new /obj/item/device/firing_pin/test_range(src)
+	new /obj/item/device/firing_pin/test_range(src)
+	new /obj/item/device/firing_pin/test_range(src)
+
+/obj/item/weapon/storage/box/loyaltypins
+	name = "box of firing pins"
+	desc = "A box of Specialised \"Loyalty\" Authentication pins produced by Nanotrasen - these check to see if the user of the gun it's installed in has been implanted with a loyalty implant. often used in ERTs."

```

Comment:
Specialized, Loyalty, and Authentication should be lowercase, often should be capitalized.